moveos/moveos/src/moveos_test_runner.rs

```diff
@@ -0,0 +1,964 @@
+            let re = Regex::new("/tmp/.[0-9a-zA-Z]+").unwrap();
+            Some(
+                re.replace_all(warn_msg.as_str(), "/tmp/tempfile")
+                    .to_string(),
+            )
+        } else {
+            None
+        }
+    };
+
+    fn rendered_diags(files: &FilesSourceText, diags: Diagnostics) -> Option<String> {
+        if diags.is_empty() {
+            return None;
+        }
+
+        let error_buffer = if read_bool_env_var(move_command_line_common::testing::PRETTY) {
+            move_compiler::diagnostics::report_diagnostics_to_color_buffer(files, diags)
+        } else {
+            move_compiler::diagnostics::report_diagnostics_to_buffer(files, diags)
+        };
+        Some(String::from_utf8(error_buffer).unwrap())
+    }
+
+    use move_compiler::PASS_COMPILATION;
+    let (mut files, comments_and_compiler_res) =
+        move_compiler::Compiler::from_files(vec![path], deps.to_vec(), named_address_mapping)
+            .set_pre_compiled_lib_opt(pre_compiled_deps)
+            .set_flags(move_compiler::Flags::empty().set_sources_shadow_deps(true))
+            .run::<PASS_COMPILATION>()?;
+    let units_or_diags = comments_and_compiler_res
+        .map(|(_comments, move_compiler)| move_compiler.into_compiled_units());
+
+    match units_or_diags {
+        Err(diags) => {
+            if let Some(pcd) = pre_compiled_deps {
+                for (file_name, text) in &pcd.files {
+                    // TODO This is bad. Rethink this when errors are redone
+                    if !files.contains_key(file_name) {
+                        files.insert(*file_name, text.clone());
+                    }
+                }
+            }
+
+            Err(anyhow!(rendered_diags(&files, diags).unwrap()))
+        }
+        Ok((mut units, warnings)) => {
+            let warnings = rendered_diags(&files, warnings);
+            let len = units.len();
+            if len != 1 {
+                panic!("Invalid input. Expected 1 compiled unit but got {}", len)
+            }
+            let unit = units.pop().unwrap();
+            let merged_output = merge_output(extended_checks_error, warnings);
+            Ok((unit, merged_output))
+        }
+    }
+}
+
+fn compile_ir_module<'a>(
+    deps: impl Iterator<Item = &'a CompiledModule>,
+    file_name: &str,
+) -> Result<CompiledModule> {
+    use move_ir_compiler::Compiler as IRCompiler;
+    let code = std::fs::read_to_string(file_name).unwrap();
+    IRCompiler::new(deps.collect()).into_compiled_module(&code)
+}
+
+fn compile_ir_script<'a>(
+    deps: impl Iterator<Item = &'a CompiledModule>,
+    file_name: &str,
+) -> Result<CompiledScript> {
+    use move_ir_compiler::Compiler as IRCompiler;
+    let code = std::fs::read_to_string(file_name).unwrap();
+    let (script, _) = IRCompiler::new(deps.collect()).into_compiled_script_and_source_map(&code)?;
+    Ok(script)
+}
+
+pub trait MoveOSMainTestAdapter<'a>: Sized {
```

Comment:
How about naming it `MoveOSTestAdapter`?

Reply:
Okay, the name has been changed.